packages/theme-default/src/components/Sidebar/index.tsx

```diff
@@ -72,14 +77,26 @@ export let matchCache: WeakMap<
 
 export function Sidebar(props: Props) {
   const { isSidebarOpen, beforeSidebar, afterSidebar, uiSwitch } = props;
+
+  const { pathname: rawPathname } = useLocation();
   const { items: rawSidebarData } = useSidebarData();
+  const [sidebarData, setSidebarData] = useState<SidebarData>(() => {
+    return rawSidebarData.filter(Boolean).flat();
+  });
+
   const localesData = useLocaleSiteData();
-  const { pathname: rawPathname } = useLocation();
-  const langRoutePrefix = normalizeSlash(localesData.langRoutePrefix || '');
-  const [sidebarData, setSidebarData] = useState<
-    (ISidebarDivider | ISidebarItem | NormalizedSidebarGroup)[]
-  >(rawSidebarData.filter(Boolean).flat());
   const pathname = decodeURIComponent(rawPathname);
+  const langRoutePrefix = normalizeSlash(localesData.langRoutePrefix || '');
+
+  useEffect(() => {
+    if (inBrowser) {
+      if (isSidebarOpen) {
+        document.body.style.overflow = 'hidden';
```

Comment:
is this body style change have side effects

Reply:
I referred to the implementation of `vitepress`, but I did not handle the special case where the overflow on the body may have an initial value.

Reply:
Every modification of the body style always brings some issues.... We need to be more cautious. Can you verify it more or can we not use the body style?

Reply:
I think this is the best way, and its impact will only exist when the sidebar is opened.

Reply:
There are some special circumstances that need to be considered and resolved.

Reply:
nothing yet